File: Tool/PreUtil/src/output/javascript/hard/SuperPalindromes.js
/**https://leetcode.com/problems/super-palindromes/ */
//Let's say a positive integer is a super-palindrome if it is a palindrome, and it is also the square of a palindrome.
//Given two positive integers left and right represented as strings, return the number of super-palindromes integers in the inclusive range [left, right].


/**
 * @param {string} left
 * @param {string} right
 * @return {number}
 */
var superpalindromesInRange = function(left, right) {
    let ans = 0;
    let l = Math.ceil(Math.sqrt(Number(left)));
    let r = Math.floor(Math.sqrt(Number(right)));
    for(let i = l; i <= r; i++){
        if(isPalindrome(i) && isPalindrome(i * i)){
            ans++;
        }
    }
    for(let i = 1; i <= 100000; i++){
        let s = i.toString();
        let t = s.split("").reverse().join("");
        let num = Number(s + t);
        if(num > r){
            break;
        }
        if(num >= l && isPalindrome(num * num)){
            ans++;
        }
    }
    return ans;
    
    
};

    

File: Tool/PreUtil/src/output/python/hard/SuperPalindromes.py
#https://leetcode.com/problems/super-palindromes/
#Let's say a positive integer is a super-palindrome if it is a palindrome, and it is also the square of a palindrome.
#Given two positive integers left and right represented as strings, return the number of super-palindromes integers in the inclusive range [left, right].


class Solution(object):
    def superpalindromesInRange(self, left, right):
        """
        :type left: str
        :type right: str
        :rtype: int
        """     
        if not left or not right:
            return 0
        if left==right:
            return 0
        if len(left)!=len(right):
            return 0
        if len(left)==1:
            return 0
        if len(left)==2:
            if left==right:
                return 0
            else:
                return 1
        # find the first different character
        for i in range(len(left)):
            if left[i]!=right[i]:
                break
        # swap the first different character
        left=left[:i]+right[i]+left[i+1:]
        return 1+self.superpalindromesInRange(left,right)
        
        
    